Scripts/RemediationScripts/Remediate-ConfigureIoTHubTLSVersion.ps1

```diff
@@ -0,0 +1,870 @@
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context)) {
+        Write-Host "Connecting to Azure account..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "***To Set MinTLSVersion on IoT Hubs in a Subscription, Contributor or higher privileges on the IoT Hubs are required.***" -ForegroundColor $([Constants]::MessageType.Warning)
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all IoT Hub(s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list to store resource details.
+    $IoTHubDetails = @()
+
+    #Required Min TLS Version
+    $requiredMinTLSVersion = 1.2
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources = @()
+    $iotHubsAPI = "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.Devices/IotHubs?api-version=2023-06-30" 
+
+    #Control id for the control
+    $controlIds = "Azure_IoTHub_DP_Use_Secure_TLS_Version"
+
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+            Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        Write-Host "Fetching all IoT Hub(s) failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $subId = $controlForRemediation.SubscriptionId
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No IoT Hub(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+                $iotHubResourceAPI = "https://management.azure.com/subscriptions/$subId/resourceGroups/$($_.ResourceGroupName)/providers/Microsoft.Devices/IotHubs/$($_.ResourceName)?api-version=2023-06-30" 
```

Comment:
didn't we have any PS command instead of calling an API?

Reply:
No, there is powershell command to update AzIotHub. But setting MinTLSVersion property with that command is not possible